### PR TITLE
owmods-cli: Add mono wrap and fix man pages install

### PR DESCRIPTION
**Pull request by loco-choco**
## Description of changes
- Added mono wrap and fix man pages install.
- Update owmods-cli [from 0.12.2 to 0.13.0](https://github.com/ow-mods/ow-mod-man/compare/gui_v0.12.2...gui_v0.13.0) ([0.13.0](https://github.com/ow-mods/ow-mod-man/releases/tag/gui_v0.13.0))
- Moved package file from applications/misc/owmods-cli/default.nix to pkgs/games/owmods-cli
- Added [Bwc9876](https://github.com/Bwc9876) and [Spoonbaker](https://github.com/Spoonbaker) as package maintainers

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -40452,7 +40452,7 @@ with pkgs;
     jre = openjdk19; # TODO: remove override https://github.com/NixOS/nixpkgs/pull/89731
   });
 
-  owmods-cli = callPackage ../applications/misc/owmods-cli {
+  owmods-cli = callPackage ../games/owmods-cli {
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:40455]: Please move the package to `pkgs/by-name/ow/owmods-cli/package.nix` and remove this line, thanks.

File: pkgs/games/owmods-cli/default.nix [hunk 2]
```diff
@@ -57,7 +64,7 @@ rustPlatform.buildRustPackage rec {
     downloadPage = "https://github.com/ow-mods/ow-mod-man/releases/tag/cli_v${version}";
     changelog = "https://github.com/ow-mods/ow-mod-man/releases/tag/cli_v${version}";
     mainProgram = "owmods";
-    license = licenses.gpl3Plus;
-    maintainers = with maintainers; [ locochoco ];
+    license = licenses.gpl3;
```
Comment (wegank) [inline pkgs/games/owmods-cli/default.nix:67]: `licenses.gpl3` is deprecated, please revert this change.

File: pkgs/games/owmods-cli/default.nix [hunk 3]
```diff
@@ -44,9 +50,10 @@ rustPlatform.buildRustPackage rec {
 
   postInstall = ''
     cargo xtask dist_cli
-    installManPage man/man*/*
+    installManPage dist/cli/man/*
     installShellCompletion --cmd owmods \
-      dist/cli/completions/owmods.{bash,fish,zsh}
+    dist/cli/completions/owmods.{bash,fish,zsh}
+    ${lib.optionalString wrapWithMono "wrapProgram $out/bin/${meta.mainProgram} --prefix PATH : '${mono}/bin'"}
```
Comment (wegank) [inline pkgs/games/owmods-cli/default.nix:56]:
```suggestion
      dist/cli/completions/owmods.{bash,fish,zsh}
  '' + lib.optionalString wrapWithMono ''
    wrapProgram $out/bin/${meta.mainProgram} --prefix PATH : '${mono}/bin'
```

File: pkgs/games/owmods-cli/default.nix [hunk 4]
```diff
@@ -6,30 +6,36 @@
...
 
   nativeBuildInputs = [
     pkg-config
     installShellFiles
-  ];
+  ] ++ lib.optional wrapWithMono makeWrapper;
 
   buildInputs = [
     zstd
+    libsoup
```
Comment (wegank) [inline pkgs/games/owmods-cli/default.nix:38]: Is `libsoup_3` OK for the package?

File: pkgs/games/owmods-cli/default.nix [hunk 5]
```diff
@@ -6,30 +6,36 @@
...
   src = fetchFromGitHub {
     owner = "ow-mods";
     repo = "ow-mod-man";
     rev = "cli_v${version}";
-    hash = "sha256-AfqpLL3cGZLKW5/BE6SaBe4S8GzYM2GKUZU8mFH5uX4=";
+    hash = "sha256-JCPuKGO0pbhQaNmZUcZ95EZbXubrjZnw0qJmKCGuAoQ=";
   };
 
-  cargoHash = "sha256-PhdfpiUgeOB13ROgzPBYM+sBLGMP+RtV9j9ebo8PpJU=";
+  doCheck = false;
```
Comment (wegank) [inline pkgs/games/owmods-cli/default.nix:28]: Why disabling the entire check instead of dropping some test cases?

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -2930,6 +2930,12 @@
     githubId = 37375448;
     name = "Buildit";
   };
+  bwc9876 = {
```
Comment (wegank) [inline maintainers/maintainer-list.nix:2933]: I would need explicit consents from @Bwc9876 and @Spoonbaker to approve this part.

File: pkgs/games/owmods-cli/default.nix (hunk 4 above)
Comment (loco-choco) [reply pkgs/games/owmods-cli/default.nix]: `libsoup_3` seems to work fine, so I will have it changed to that.

File: pkgs/games/owmods-cli/default.nix (hunk 5 above)
Comment (loco-choco) [reply pkgs/games/owmods-cli/default.nix]: Not really sure why it was set to false to be honest, but allowing it to be true isn't resulting in any issues, so I guess there isn't a problem in removing that line.

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (Spoonbaker) [reply maintainers/maintainer-list.nix:2933]: I am cool with this.

File: pkgs/by-name/ow/owmods-cli/package.nix [hunk 7]
```diff
@@ -6,34 +6,39 @@
...
+  ] ++ lib.optional wrapWithMono makeWrapper;
 
   buildInputs = [
     zstd
+    libsoup_3
   ] ++ lib.optionals stdenv.isLinux [
     openssl
   ] ++ lib.optionals stdenv.isDarwin [
-    Security
+    darwin.Security
```
Comment (wegank) [inline pkgs/by-name/ow/owmods-cli/package.nix:41]:
```suggestion
    darwin.apple_sdk.frameworks.Security
```

`darwin.Security` looks like a questionable alias, I'm not sure if it would last long in Nixpkgs.

File: maintainers/maintainer-list.nix (hunk 6 above)
Comment (Bwc9876) [reply maintainers/maintainer-list.nix:2933]: I am also cool with this.

**Review by Bwc9876 — APPROVED** [review]

**Review by Spoonbaker — APPROVED** [review]
